lib/kafo/puppet_module.rb

```diff
@@ -25,7 +25,7 @@ def self.find_parser
       end
     end
 
-    def initialize(identifier, parser = self.class.find_parser, configuration = KafoConfigure.config)
```

Comment:
Perhaps it should keep the parser argument but rather set it to `nil`. Then `if @parser.nil?` it calls `self.class.find_parser`. Since `find_parser` returns `:none` if it isn't found, this should be ok and keep the tests working.

Reply:
Updated with this suggestion and moved it out of draft